src/bmunit.c

```diff
@@ -989,7 +989,7 @@ void UnitChangeFaction(struct Unit* unit, int faction) {
         GetUnit(newUnit->rescue)->rescue = newUnit->index;
 }
 
-inline s8 CanUnitCrossTerrain(struct Unit* unit, int terrain) {
+inline s32 CanUnitCrossTerrain(struct Unit* unit, int terrain) {
```

Comment:
This is likely another bug in the official code. Theoretically, there should only be a version of the function with a return type of s8. However, under the previous approach—where the function was not declared in the header file (.h) but individually declared in each source file (.c)—modern gcc would fail to link, unable to find the symbol for this function.

The current modification, while not ideal, ensures both the binary consistency after compilation with agbcc and resolves the linking issue with modern gcc.